plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,268 @@
+package plugins
+
+import (
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/openlyinc/pointy"
+	"github.com/stellar/kelp/queries"
+	"github.com/stellar/kelp/support/utils"
+
+	"github.com/stellar/go/txnbuild"
+
+	hProtocol "github.com/stellar/go/protocols/horizon"
+	"github.com/stellar/kelp/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func makeTestVolumeFilterConfig(baseCapInBase, baseCapInQuote float64, additionalMarketIDs, optionalAccountIDs []string, mode volumeFilterMode) *VolumeFilterConfig {
+	var baseCapInBasePtr *float64
+	if baseCapInBase >= 0 {
+		baseCapInBasePtr = pointy.Float64(baseCapInBase)
+	}
+
+	var baseCapInQuotePtr *float64
+	if baseCapInQuote >= 0 {
+		baseCapInQuotePtr = pointy.Float64(baseCapInQuote)
+	}
+
+	return &VolumeFilterConfig{
+		SellBaseAssetCapInBaseUnits:  baseCapInBasePtr,
+		SellBaseAssetCapInQuoteUnits: baseCapInQuotePtr,
+		mode:                         mode,
+		additionalMarketIDs:          additionalMarketIDs,
+		optionalAccountIDs:           optionalAccountIDs,
+	}
+}
+
+func makeWantVolumeFilter(config *VolumeFilterConfig, firstMarketID string, marketIDs []string, optionalAccountIDs []string, action string) *volumeFilter {
+	queryMarketIDs := utils.Dedupe(append([]string{firstMarketID}, marketIDs...))
+	query, e := queries.MakeDailyVolumeByDateForMarketIdsAction(&sql.DB{}, queryMarketIDs, action, optionalAccountIDs)
+	if e != nil {
+		panic(e)
+	}
+
+	return &volumeFilter{
+		name:                   "volumeFilter",
+		baseAsset:              utils.NativeAsset,
+		quoteAsset:             utils.NativeAsset,
+		config:                 config,
+		dailyVolumeByDateQuery: query,
+	}
+}
+
+func TestMakeFilterVolume(t *testing.T) {
+	testAssetDisplayFn := model.MakeSdexMappedAssetDisplayFn(map[model.Asset]hProtocol.Asset{model.Asset("XLM"): utils.NativeAsset})
+	configValue := ""
+	exchangeName := ""
+	tradingPair := &model.TradingPair{Base: "XLM", Quote: "XLM"}
+	modes := []volumeFilterMode{volumeFilterModeExact, volumeFilterModeIgnore}
+	firstMarketID := MakeMarketID(exchangeName, "native", "native")
+
+	testCases := []struct {
+		name       string
+		marketIDs  []string
+		accountIDs []string
+		wantFilter *volumeFilter
+	}{
+		// TODO DS Confirm the empty config fails once validation is added to the constructor
+		{
+			name:       "1 market id",
+			marketIDs:  []string{"marketID"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 market ids",
+			marketIDs:  []string{"marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "2 dupe market ids, 1 distinct",
+			marketIDs:  []string{"marketID1", "marketID1", "marketID2"},
+			accountIDs: []string{},
+		},
+		{
+			name:       "1 account id",
+			marketIDs:  []string{},
+			accountIDs: []string{"accountID"},
```

Comment:
wantMarketIDs should be a non-empty string array here I think (would be the string inlined value in `firstMarketID`)?